From 4216e8b1da2d64b8e3ad371514387b9297b27e9b Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=C3=86var=20Arnfj=C3=B6r=C3=B0=20Bjarmason?=
 <avarab@gmail.com>
Date: Wed, 23 May 2018 15:15:37 +0000
Subject: [PATCH 2/2] v2.17.1: make the RelNotes understandable & update docs
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

The release notes very vaguely referred to "has support", but didn't
make it obvious to someone not looking at the patches how to actually
enable this, and we didn't add anything to the documentation
either. Fix both of those.

Signed-off-by: Ævar Arnfjörð Bjarmason <avarab@gmail.com>
---
 Documentation/RelNotes/2.17.1.txt | 12 +++++++-----
 Documentation/config.txt          |  4 ++++
 2 files changed, 11 insertions(+), 5 deletions(-)

diff --git a/Documentation/RelNotes/2.17.1.txt b/Documentation/RelNotes/2.17.1.txt
index 78dd47c820..2913729576 100644
--- a/Documentation/RelNotes/2.17.1.txt
+++ b/Documentation/RelNotes/2.17.1.txt
@@ -15,11 +15,13 @@ Fixes since v2.17
    "/../" and ".git/" etc., is now ignored.
 
  * In addition to the above fix that also appears in maintenance
-   releases v2.13.7, v2.14.4, v2.15.2 and v2.16.4, this has support on
-   the server side to reject pushes to repositories that attempt to
-   create such problematic .gitmodules file etc. as tracked contents,
-   to help hosting sites protect their customers by preventing
-   malicious contents from spreading.
+   releases v2.13.7, v2.14.4, v2.15.2 and v2.16.4, this release
+   extends transfer.fsckObjects (off by default) to reject fetches or
+   pushes to repositories that attempt to create such problematic
+   .gitmodules file etc. as tracked contents, to help hosting sites
+   protect their customers by preventing malicious contents from
+   spreading, or to protect clients that fetch from passing on a bad
+   repository to their downstream fetchers.
 
 Credit for finding this vulnerability and the proof of concept from
 which the test script was adapted goes to Etienne Stalmans.  Credit
diff --git a/Documentation/config.txt b/Documentation/config.txt
index 28f84db492..a1cd192782 100644
--- a/Documentation/config.txt
+++ b/Documentation/config.txt
@@ -3273,6 +3273,10 @@ transfer.fsckObjects::
 When set, the fetch or receive will abort in the case of a malformed
 object or a broken link. The result of an abort are only dangling
 objects.
++
+This will also check that the '.gitmodules' file isn't malicious,
+which was a security issue initially fixed in v2.17.1, and other such
+sanity checks might be added in the future.
 
 transfer.hideRefs::
 	String(s) `receive-pack` and `upload-pack` use to decide which
-- 
2.17.0.290.gded63e768a

